pkg/fuzzer/queue/queue.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package queue
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+
+	"github.com/google/syzkaller/pkg/hash"
+	"github.com/google/syzkaller/pkg/ipc"
+	"github.com/google/syzkaller/pkg/signal"
+	"github.com/google/syzkaller/pkg/stats"
+	"github.com/google/syzkaller/prog"
+)
+
+type Request struct {
+	Prog       *prog.Prog
+	NeedSignal SignalType
+	NeedCover  bool
+	NeedHints  bool
+	// TODO: passing ExecOpts directly is actually quite hacky -- this is a high-level interface
```

Comment:
I am not sure we can have significantly higher-level interface. The flags we have in Request are mostly match one-to-one to ipc flags. These are just separate bools vs bits, but that looks more like a style preference than actual abstraction level.

The downside of duplicating these flags in another interface looks much worse.
That's the same problem you mentioned as well: we have host.features, csource options, csource flags, ipc flags and all effectively duplicate each other and have tons of code that that converts between these, and that code is constant source of bugs (I fixed at least 2 just last week).

So instead of introducing yet another duplicate of these flags in a different form and adding yet more conversions, I would rather converge on a single version and use it though out the system.

Result already contains ipc.ProgInfo. You could convert it as well to higher level representation, but I think it works just fine. If we have ipc.ProgInfo in Result, it's reasonable to have ipc.ExecOpts in Requst.